src/System.Text.JsonLab/System/Text/Json/JsonWriter.cs

```diff
@@ -125,32 +125,84 @@ public void WriteObjectStart(string name)
         {
             ReadOnlySpan<byte> nameSpan = MemoryMarshal.AsBytes(name.AsSpan());
 
+            if (_prettyPrint)
+            {
+                WriteStartUtf8Pretty(nameSpan, JsonConstants.OpenBrace);
+            }
+            else
+            {
+                while (!TryWriteStartUtf8(nameSpan, JsonConstants.OpenBrace))
+                    EnsureBuffer();
+            }
+
+            _indent &= RemoveFlagsBitMask;
+            _indent++;
+
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
```

Comment:
You do not need to mark this with NoInlining. It would be insane for JIT to voluntarily inline method that is this big.